streaming/base/format/mds/reader.py

```diff
@@ -120,4 +120,8 @@ def get_sample_data(self, idx: int) -> bytes:
             begin, end = np.frombuffer(pair, np.uint32)
             fp.seek(begin)
             data = fp.read(end - begin)
+        if not data:
```

Comment:
While working on a test case, I found out that if the sample index is greater than it should be in a shard file, then `data` ends up being empty byte `b''` and `decode_sample()` methods fails with `ValueError: not enough values to unpack (expected 1, got 0)`. Added a check here, so that we have a proper error message. Now, it says `IndexError: Relative sample index 3 is not present in the shard.00000.mds file.` which is helpful to debug.